### ISSUE TITLE
Generic.Arrays.ArrayIndent has conflicting logic

### ISSUE THREAD

**Issue by morozov** [squizlabs/PHP_CodeSniffer#3100]
**Describe the bug**
The fix for #2882 (04253420ca203acde87d06ff122f3c5e08b043f9) introduces conflicting logic in the `Generic.Arrays.ArrayIndent` sniff.

**Code sample**
```php
<?php

public function test()
{
    return [
        [
            'foo'
                . 'bar',
            [
                    'baz',
                    'qux',
                ],
        ],
    ];
}
```

**Custom ruleset**
None

**To reproduce**
1. Check the file above by running `phpcs -s --standard=Squiz --sniffs=Generic.Arrays.ArrayIndent test.php`:
   ```
   FILE: /home/morozov/Projects/phpcs-playground/test.php
   ------------------------------------------------------------------------------------------------
   FOUND 1 ERROR AFFECTING 1 LINE
   ------------------------------------------------------------------------------------------------
    9 | ERROR | [x] Array open brace not indented correctly; expected at least 16 spaces but found
      |       |     12 (Generic.Arrays.ArrayIndent.OpenBraceIncorrect)
   ------------------------------------------------------------------------------------------------
   PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
   ------------------------------------------------------------------------------------------------
   
   Time: 46ms; Memory: 8MB
   ```
2. Try fixing the violation manually:
   ```diff
   --- a/test.php
   +++ b/test.php
   @@ -6,7 +6,7 @@ public function test()
            [
                'foo'
                    . 'bar',
   -            [
   +                [
                        'baz',
                        'qux',
                    ],
   ```
3. Check the file again by running `phpcs -s --standard=Squiz --sniffs=Generic.Arrays.ArrayIndent test.php`:
   ```
   FILE: /home/morozov/Projects/phpcs-playground/test.php
   ------------------------------------------------------------------------------------------------
   FOUND 1 ERROR AFFECTING 1 LINE
   ------------------------------------------------------------------------------------------------
    9 | ERROR | [x] Array key not indented correctly; expected 12 spaces but found 16
      |       |     (Generic.Arrays.ArrayIndent.KeyIncorrect)
   ------------------------------------------------------------------------------------------------
   PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
   ------------------------------------------------------------------------------------------------
   
   Time: 48ms; Memory: 8MB
   ```
4. Try auto-fixing the file by running `phpcbf --standard=Squiz --sniffs=Generic.Arrays.ArrayIndent test.php`:
   ```
   PHPCBF RESULT SUMMARY
   ----------------------------------------------------------------------
   FILE                                                  FIXED  REMAINING
   ----------------------------------------------------------------------
   /home/morozov/Projects/phpcs-playground/test.php      FAILED TO FIX
   ----------------------------------------------------------------------
   A TOTAL OF 0 ERRORS WERE FIXED IN 1 FILE
   ----------------------------------------------------------------------
   PHPCBF FAILED TO FIX 1 FILE
   ----------------------------------------------------------------------
   
   Time: 116ms; Memory: 8MB
   ```

**Expected behavior**
There's no conflict, and the file is auto-fixed.

**Versions (please complete the following information):**
 - OS: Linux
 - PHP: 7.4.9
 - PHPCS: 04253420ca203acde87d06ff122f3c5e08b043f9
 - Standard: Squiz
